omegatrade-app/frontend/src/app/components/company/manage-company/manage-company.component.ts

```diff
@@ -0,0 +1,92 @@
+import { Component, ViewChild } from '@angular/core';
+import { MatSort } from '@angular/material/sort';
+import { MatPaginator } from '@angular/material/paginator';
+import { MatTableDataSource } from '@angular/material/table';
+import { MatDialog } from '@angular/material/dialog';
+import { UpdateCompanyComponent } from '../update-company/update-company.component';
+import { RestService } from '../../../services/rest.service';
+import { SnackBarService } from '../../../services/snackbar.service';
+import { take } from "rxjs/operators";
+
+@Component({
+  selector: 'app-manage-company',
+  templateUrl: './manage-company.component.html',
+  styleUrls: ['./manage-company.component.css']
+})
+
+export class ManageCompanyComponent {
+  displayedColumns: string[] = ['companyName', 'companyShortCode', 'action'];
+  dataSource: MatTableDataSource<CompanyData>;
+  @ViewChild(MatPaginator, { static: true }) paginator: MatPaginator;
+  @ViewChild(MatSort) sort: MatSort;
+
+  response: any;
+  loader: boolean = false;
+
+  constructor(private snackBarService: SnackBarService, private restService: RestService, public dialog: MatDialog) {
+    this.getCompanies();
```

Comment:
Instead of calling getCompanies function in the constructor, it should be called inside ngOnInit hook to use the component life cycle in better way

Reply:
this issue has been resolved.